### ISSUE TITLE
A11y_AzureToolsStorageExplorer_ConfigureSoftDeletePolicy_ScreenReader: NVDA announces entire content present in screen when we tab to shift NVDA focus to "What is Soft Delete?" link.

### ISSUE THREAD

**Issue by Suraj1200499** [microsoft/AzureStorageExplorer#1908]
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags: #A11y_AzureToolsStorageExplorer; #A11yMAS; #A11YTCS; #DesktopApp; #MAS1.3.1; #ScreenReader; #NVDA; #JAWS; #Win32; 

**Environment Details:**
Application Name: Azure Tools Storage Explorer
Application Version: 1.10.0
Windows10

**Repro Steps:**
1. Launch Storage Explorer
2. Navigate to "Open Connect dialog" pane and Sign in to Azure.
3. Go to "Manage Accounts" pane and Select the subscriptions you will use. Once selected, click on "Apply" button.
4. Navigate to "Blob Containers" of the selected subscription, right click and Select "Configure Soft Delete Policy" and hit enter.
5. Start NVDA.
6. Navigate to "What is Soft Delete?" link.
7. Verify NVDA announces entire content present in screen when we tab to shift NVDA focus to "What is Soft Delete?" link.

**​Actual:​**
**Issue1:** While navigating on the screen using Tab key for the first time, NVDA announces entire content present in screen when we tab to shift NVDA focus to "What is Soft Delete?" link.  
* Same issue is observed for "Enable Soft Delete" radio button in this Scenario.
Same issue is observed in "**JAWS**" also.

**Issue2:** And also for second time While navigating on the screen using Tab key, NVDA announces "dialog cancel button What is Soft Delete?" when NVDA focus is on "What is Soft Delete?".
Same issue is observed in "**JAWS**" also.

**Same Issue1 is observed in below Scenarios:**
1. **SAS URL**
Here in this Scenario NVDA announces entire screen content when Shared Access Signature screen gets opened.

**Same Issue2 is observed in below Scenarios:**
1. **Delete Popup**
Here in this Scenario NVDA announces dialog title when shift focus from delete control to cancel control.
2. **Rename blob**
Here in this Scenario NVDA announces dialog title when shift focus from New name edit box to Rename control.
3. **Update Access Tier**
Here in this Scenario NVDA announces entire screen content when Press Tab key to shift focus to "Learn More" link.
Same issue is observed in "**JAWS**" also.
4. **Set Container Public Access Level**
Here in this Scenario NVDA unnecessary announces other content present on the screen when Tab to shift keyboard focus to "No Public Access" radio button.
Same issue is observed in "**JAWS**" also.
5. **Undelete Selected**
Here in this Scenario NVDA unnecessary announces other content present on the screen when Tab to shift keyboard focus to "Cancel" control for the first time.

**Expected:**
**Issue1:** NVDA should not announce entire content present in screen when we tab to shift NVDA focus to "What is Soft Delete?" link, Instead NVDA should announce only link name with description.
**Issue2:** And also for second time While navigating on the screen using Tab key, NVDA should not announce" dialog cancel button What is Soft Delete?" when NVDA focus is on "What is Soft Delete?", Instead NVDA should announce only link name with description.

**UserImpact:**
Screen reader users gets confusion if screen reader start announcing entire content of the screen at a time when shift screen reader focus to link.

**Recommendation:**
https://microsoft.sharepoint.com/teams/msenable/mas/Pages/browse-fixes.aspx

**MAS Reference:**
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

**Attachment for Reference:**
[A11y_AzureToolsStorageExplorer_ConfigureSoftDeletePolicy_ScreenReader.pptx](https://github.com/microsoft/AzureStorageExplorer/files/3651903/A11y_AzureToolsStorageExplorer_ConfigureSoftDeletePolicy_ScreenReader.pptx)

**Comment by JasonYeMSFT** [comment]
@Suraj1200499, I can confirm this bug on NVDA. But I also tried Jaws and cannot reproduce this issue there. The screen reader doesn't announce the whole page when navigating using Jaws. Could you please show me a recording with the speech history from Jaws to let me see the exact repro steps for Jaws?

**Comment by JasonYeMSFT** [comment]
This is the default behavior of NVDA. NVDA enters a web page in browse mode by default. It will read the entire page the first time it gets loaded. There is an option for this "Say all on loaded" behavior and is turned on by default.

The reason that some dialog doesn't have this issue is that we gave the initial focus to an element which is considered to be a control element by NVDA. In this case, NVDA automatically enters the focus mode and don't read the entire page. 

Ryan Shugart, the accessibility expert suggest us not to change the default behavior of the screen reader. We are going to mark this issue (and the equivalent ones for other screen readers) as won't fix.

**Comment by Suraj1200499** [comment]
@JasonYeMSFT 
The bug is By design, As per our process By design bug will be validated by our SME's. So I had put this bug in their note. Once I get a reply from them, I will update the comment.

**Comment by Suraj1200499** [comment]
We got reply from SME to close the bug.
Hence Closing the bug.

**Comment by fsteffi** [comment]
GithubTags:#Closed;

**Comment by mstechie** [comment]
GitHubTags:#BM-TCS-StorageExplorer-Win32-Sep2019;
